docs/dcp2_system_design.rst

```diff
@@ -0,0 +1,1465 @@
+updates must be written to an object under a new name, with a
+different—usually higher, rarely lower—value for ``version``. The
+importer places all objects from the staging area into the TDR dataset,
+even if the resulting raw table rows don't end up being referenced by
+the snapshot for the major release currently in preparation. A minor
+release snapshot might refer to it.
+
+An object can be removed from the staging area but the removal of an
+object will not lead to the deletion of the corresponding table row from
+the TDR dataset or from any snapshots made of that dataset. In order to
+exclude data or metadata from a subsequently created snapshot of the TDR
+dataset, special deletion objects [TODO: reference section] must be
+created in the staging area. When removing an object from the staging
+area, all objects referring to it must also be removed so that the
+referential integrity of the staging area is maintained, except when the
+referred object has already been imported. [This is rather complicated.]
+
+WIP: Updates to metadata entities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To update a metadata entity, the source creates a new object in the
+staging area. The name of that object contains the ``entity_id`` of the
+entity to be updated and a ``version`` that is strictly higher than the
+``version`` of the TDR dataset row this update makes obsolete. The new
+``version`` does not necessarily need to be set to the current time or
+any other maximal value. The only requirement is that it is greater than
+the obsoleted version in TDR. Back-dating metadata updates like that is
+necessary for minor releases.
+
+If the updated metadata entity describes a data file, the property
+holding the `path of the corresponding data file <Data file paths_>`_
+must not be modified and the data file object should be omitted from the
+staging area. The TDR importer detects this case and modifies the
+corresponding row in the TDR table but reuses the existing data file in
+Firestore [I might be using the wrong terminology here]. This is a
+direct consequence of the `importer's idempotence <Importer idempotence_>`_.
+If the referenced data file does not exist in Firestore, the importer
+`records an error <Import errors_>`_. There is a separate section
+[TODO: reference section] dedicated to data files updates.
+
+After the source applies this process to all entities it intends to
+update, it requests the importer to be invoked.
+
+Notes from comments:
+
+-  Cases for DELETE markers:
+
+   -  Impractical to stage Tabula Muris except one object only to delete
+      that object.
+
+   -  Deletions should be explicit actions, rather than a potentially
+      accidental absence of an object.
+
+-  Delta staging area partiality:
+
+   a) omit any object that hasn't change
+
+   b) (stricter) require that subgraphs are fully defined
+
+   c) like b but allow omission of data files
+
+   d) like c but also allow omission of descriptors
```

Comment:
`a)` `b)` `c)` `d)` display as `i.` `ii.` `iii.` `iv.` making the references `like b` and `like c` invalid.

Reply:
I can't figure out how to control the numbering style; I don't think it's possible. I changed these to auto-number and changed the text to say "like previous, but..."